### ISSUE TITLE
background-geolocation: Unhandled promise rejection when permissions are disabled (in iOS)

### ISSUE THREAD

**Issue by bionicbrad** [driftyco/ionic-native#1118]
If I have an app using the background geolocation native plugin, when I disable permissions in the iOS settings, the BackgroundGeolocation.configure() method fails, and doesn't call back to the error handler as expected.  Here's what I see being logged in Xcode:

```
2017-02-26 15:36:20.033007 app[4831:1027795] LocationManager will start: 0
2017-02-26 15:36:20:008 app[4831:1027732] LocationManager will start: 0
2017-02-26 15:36:20.067317 app[4831:1027663] ERROR: Unhandled Promise rejection: [object Object] ; Zone: <root> ; Task: setTimeout ; Value: [object Object]
2017-02-26 15:36:20.067634 app[4831:1027663] ERROR: error JSON.stringify()ing argument: TypeError: JSON.stringify cannot serialize cyclic structures.
```

From what I can tell, the underlying cordova plugin is reporting the error as expected, where it reaches the line that returns the CDVCommandStatus_ERROR with the error info:

```objc
- (void) start:(CDVInvokedUrlCommand*)command
{
    [self.commandDelegate runInBackground:^{
        NSError *error = nil;
        CDVPluginResult* result = nil;

        [manager start:&error];
        if (error == nil) {
            result = [CDVPluginResult resultWithStatus:CDVCommandStatus_OK];
        } else {
            result = [CDVPluginResult resultWithStatus:CDVCommandStatus_ERROR messageAsDictionary:[error userInfo]];
        }
        
        [self.commandDelegate sendPluginResult:result callbackId:command.callbackId];
    }];
}
```

I'm working with ionic-native 2.4.1.  I'm not sure where to look beyond this...I have yet to figure out how to trace what's going on in the ionic-native module between the cordova plugin and the TypeScript code in my ionic app, but I think it's something in the ionic-native module that is perhaps not handling this error...

Any suggestions?  I'd like to be able to capture this event so that I can appropriately prompt the user to enable permissions for location services.

I don't know if the same problem exists in Android...I'll test that as soon as I have a chance.

EDIT: I just upgraded to ionic-native 2.5.1, and the problem persists...

**Comment by bionicbrad** [comment]
Thanks - though if I upgrade to 2.7.0, I get an error saying that "Supplied parameters do not match any signature of call target." associated with the BackgroundGeolocation.configure() method.  What would I need to do differently to make use of this?

**Comment by ihadeed** [comment]
@bionicbrad Upgrade to 2.8.0 and let me know if you're experiencing issues.

**Comment by bionicbrad** [comment]
No luck - ionic build ios now produces:

```
module.js:471
    throw err;
    ^

Error: Cannot find module '../dist/index'
    at Function.Module._resolveFilename (module.js:469:15)
    at Function.Module._load (module.js:417:25)
    at Module.require (module.js:497:17)
    .....
```

I think I may just have to rebuild the project from scratch.

**Comment by Scuttman** [comment]
I am getting the same issue with building android. I updated ionic-native to solve a different problem but now get:

13:47:43]  typescript: LOCALURL/src/providers/location-tracker.ts, line: 30
            Supplied parameters do not match any signature of call target.

      L30:      BackgroundGeolocation.configure((location) => {                                                                                                                                            [13:47:43]  transpile failed
      [13:47:43]  ionic-app-script task: "build"
  [13:47:43]  Error: Error
The background location is setup exactly as Josh Morony outlines in his tutorial and this used to work but I tried a clean build of [his tutorial](https://www.joshmorony.com/adding-background-geolocation-to-an-ionic-2-application/) and get errors on that too now. Any ideas? I was about to release an app so this is a massive fustration! Any help would be very welcome! Thanks in advance
